samples/matter/common/src/diagnostic/diagnostic_logs_retention.cpp

```diff
@@ -62,6 +63,9 @@ CHIP_ERROR DiagnosticLogsRetention::PushLog(const void *data, size_t size)
 	VerifyOrReturnError(mIsInitialized, CHIP_ERROR_INCORRECT_STATE);
 	VerifyOrReturnError(size < mCapacity, CHIP_ERROR_BUFFER_TOO_SMALL);
 
+	/* Make sure that no-one will start write operation from different thread in the same time. */
+	k_mutex_lock(&sWriteMutex, K_FOREVER);
```

Comment:
Should we wait forever, or give some time to execute, and return?

Reply:
Discussed f2f